tool/templates/cpp/runtime.hpp.jinja

```diff
@@ -167,6 +168,64 @@ public:
       return result<T2, E>(Ok<T2>(std::move(t)));
     }
   }
+  
+  template<typename T2, typename U = T, typename V = E, typename std::enable_if_t<!std::is_same_v<U, std::monostate> && !std::is_same_v<V, std::monostate>, std::nullptr_t> = nullptr>
```

Comment:
thought: I think this should be a helper function; I don't want internal capi stuff on the `result` public API.